File: CandyCalories.java
import java.util.*;
import java.io.*;

class triple implements Comparable<triple>{
    public String str;
    public int cnt, val;
    public triple(String s, int c, int v) {
        str = s;
        cnt = c;
        val = v;
    }
    public int compareTo(triple other) {
        return Integer.compare(this.val, other.val);
    }
}

public class CandyCalories {
    public static void main(String[] args) throws FileNotFoundException{
        Scanner sc = new Scanner(new File("CandyCalories.in"));
        int n = sc.nextInt();
        sc.nextLine();
        TreeSet<triple> l = new TreeSet<triple>();
        while (n-- > 0) {
            String s = sc.next();
            int i1 = sc.nextInt();
            int i2 = sc.nextInt();
            l.add(new triple(s, i1, i2));
        }
        //TreeSet already sorted
        int cnt = 0;
        int sm = 0;
        for (triple i : l) {
            if (sm < 1500 && sm + i.val < 1500) {
                int tmp = Math.min(i.cnt, (1499 - sm) / i.val);
                sm += tmp * i.val;
                cnt += tmp;
            }
        }
        System.out.println(cnt);
    }
}

File: CandyCalories.py
import sys
sys.stdin = open("CandyCalories.in", "r")

n: int = int(input())
l: list[tuple[str, int, int]] = []

for i in range(n):
    s: str
    cnt: int
    val: int
    s, cnt, val = input().split()
    cnt, val = map(int, [cnt, val])
    l.append((s, cnt, val))

l.sort(key = lambda x: x[2])

cnt: int = 0
sm: int = 0
for i in l:
    if sm < 1500 and sm + i[2] < 1500:
        tmp: int = min(i[1], (1499 - sm) // i[2])
        sm += tmp * i[2]
        cnt += tmp
        
print(cnt)